src/js/providers/tracks-mixin.js

```diff
@@ -97,28 +98,38 @@ define(['../utils/underscore',
         }
 
         if (this._renderNatively) {
-            this.addTracksListener(this.video.textTracks, 'change', textTrackChangeHandler.bind(this));
+            // Only bind and set this.textTrackChangeHandler once so that removeEventListener works
+            this.textTrackChangeHandler = this.textTrackChangeHandler || textTrackChangeHandler.bind(this);
+
+            this.removeTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
+            this.addTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
         }
 
         if (this._textTracks.length) {
             this.trigger('subtitlesTracks', {tracks: this._textTracks});
         }
     }
 
-    function setupSideloadedTracks(tracks) {
+    function setupSideloadedTracks(itemTracks) {
+        // Determine if the tracks are the same and the embedded + sideloaded count = # of tracks in the controlbar
+        var alreadyLoaded = itemTracks === this._itemTracks;
+        if (!alreadyLoaded) {
+            _cancelXhr(this._itemTracks);
+        }
+        this._itemTracks = itemTracks;
+
         this._renderNatively = _nativeRenderingSupported(this.getName().name);
-        if (this.isSDK || !tracks) {
+        if (!itemTracks || !itemTracks.length) {
             return;
         }
 
-        if (!_tracksAlreadySideloaded.call(this, tracks)) {
+        if (!alreadyLoaded) {
```

Comment:
They were the reason there were so many side effects and are no longer necessary since `this._itemTracks = null` was added to clearTracks.